educator/src/Dscp/Educator/DB/Instances.hs

```diff
@@ -114,19 +113,22 @@ instance HasSqlValueSyntax SqliteValueSyntax TxBlockIdx where
 -- 'FromBackendRow' instances
 ----------------------------------------------------------------------------
 
+#define GenFromBackendRow(TYPE) \
+instance (BeamBackend be, BackendFromField be (TYPE)) => FromBackendRow be (TYPE)
+
 -- For SQLite they all refer to 'FromField' instances
-instance FromBackendRow Sqlite (Hash a)
-instance FromBackendRow Sqlite Address
-instance FromBackendRow Sqlite Course
-instance FromBackendRow Sqlite AssignmentType
-instance FromBackendRow Sqlite Subject
-instance FromBackendRow Sqlite Grade
-instance FromBackendRow Sqlite BlockIdx
-instance FromBackendRow Sqlite TxBlockIdx where
-instance FromBackendRow Sqlite SubmissionWitness where
-instance FromBackendRow Sqlite (MerkleSignature a) where
-instance FromBackendRow Sqlite (EmptyMerkleTree a) where
-instance FromBackendRow Sqlite ATGDelta where
+GenFromBackendRow(Hash a)
+GenFromBackendRow(Address)
+GenFromBackendRow(Course)
+GenFromBackendRow(AssignmentType)
+GenFromBackendRow(Subject)
+GenFromBackendRow(Grade)
+GenFromBackendRow(BlockIdx)
+GenFromBackendRow(TxBlockIdx)
+GenFromBackendRow(SubmissionWitness)
+GenFromBackendRow(MerkleSignature a)
+GenFromBackendRow(EmptyMerkleTree a)
+GenFromBackendRow(ATGDelta)
```

Comment:
Too bad we can't `for [] $ \ty -> ...` these.

Reply:
Maybe it's really worth switching to TH instead of macroses.
_What if_ `forM` really works at top level for TH :open_mouth:

Reply:
@Martoon-00 , `for` actually _can_ be used there, however, additional actions are required